src/_ttconv.cpp

```diff
@@ -5,157 +5,76 @@
 
   Python wrapper for TrueType conversion library in ../ttconv.
  */
-#define PY_SSIZE_T_CLEAN
 #include "mplutils.h"
 
-#include <Python.h>
+#include <pybind11/pybind11.h>
 #include "ttconv/pprdrv.h"
-#include "py_exceptions.h"
 #include <vector>
-#include <cassert>
 
 /**
  * An implementation of TTStreamWriter that writes to a Python
  * file-like object.
  */
 class PythonFileWriter : public TTStreamWriter
 {
-    PyObject *_write_method;
+  pybind11::function _write_method;
 
   public:
-    PythonFileWriter()
-    {
-        _write_method = NULL;
-    }
-
-    ~PythonFileWriter()
-    {
-        Py_XDECREF(_write_method);
-    }
-
-    void set(PyObject *write_method)
-    {
-        Py_XDECREF(_write_method);
-        _write_method = write_method;
-        Py_XINCREF(_write_method);
-    }
+    PythonFileWriter(pybind11::object& file_object)
+    : _write_method(file_object.attr("write")) {}
 
     virtual void write(const char *a)
     {
-        PyObject *result = NULL;
-        if (_write_method) {
-            PyObject *decoded = NULL;
-            decoded = PyUnicode_DecodeLatin1(a, strlen(a), "");
-            if (decoded == NULL) {
-                throw py::exception();
-            }
-            result = PyObject_CallFunctionObjArgs(_write_method, decoded, NULL);
-            Py_DECREF(decoded);
-            if (!result) {
-                throw py::exception();
-            }
-            Py_DECREF(result);
+        PyObject* decoded = PyUnicode_DecodeLatin1(a, strlen(a), "");
+        if (decoded == NULL) {
+            throw pybind11::error_already_set();
         }
+        _write_method(pybind11::handle(decoded));
+        Py_DECREF(decoded);
     }
 };
 
-int fileobject_to_PythonFileWriter(PyObject *object, void *address)
-{
-    PythonFileWriter *file_writer = (PythonFileWriter *)address;
-
-    PyObject *write_method = PyObject_GetAttrString(object, "write");
-    if (write_method == NULL || !PyCallable_Check(write_method)) {
-        PyErr_SetString(PyExc_TypeError, "Expected a file-like object with a write method.");
-        return 0;
-    }
-
-    file_writer->set(write_method);
-    Py_DECREF(write_method);
-
-    return 1;
-}
-
-int pyiterable_to_vector_int(PyObject *object, void *address)
+static void convert_ttf_to_ps(
+    const char *filename,
+    pybind11::object &output,
+    int fonttype,
+    pybind11::iterable glyph_ids)
```

Comment:
Just make this argument a std::vector<int> (probably a `std::optional<std::vector<int>>`, in fact) and don't bother with the conversion?

Reply:
With the default value of `None`, I think this needs to be `pybind11::iterable*`, to account for the `nullptr`. If we need to use `std::vector<int>`, then we need `std::optional`, which requires `C++17`:

XREF: The note section from: https://pybind11.readthedocs.io/en/stable/advanced/functions.html#allow-prohibiting-none-arguments